lib/generator.ts

```diff
@@ -542,22 +570,83 @@ export class Generator extends Cli {
     return source
   }
 
+  private sourceLines(source: string) {
+    source = source.replace(/\r\n/g, '\n')
+    source = source.replace(/\n\r/g, '\n')
+    source = source.replace(/\r/g, '\n')
+
+    const lines = source.split('\n')
+    return lines
+  }
+
+  private findDependencies(
+    typings: IDeclarationMap,
+    moduleName: string,
+    shakenTypings: Set<string>,
+    filter: 'exportOnly' | 'allImports',
+  ): void {
+    shakenTypings.add(moduleName)
+    const current = typings[moduleName]
+    const lines = this.sourceLines(current)
+    const refs =
+      filter === 'allImports'
+        ? this.findAllImportedRefs(lines)
+        : this.findExportedRefsOnly(lines)
+    verbose(`"${moduleName}" references ${JSON.stringify(refs)}`)
+
+    for (const ref of refs) {
+      this.findDependencies(typings, ref, shakenTypings, filter)
+    }
+  }
+
+  private findExportedRefsOnly(lines: string[]) {
+    const exports = /export .*? from ['"]([^'"]+)['"]/
```

Comment:
This is not a robust way to detect exports. Problem occurs due to the fact that you only analyse 1 line but expect whole expression to be present in it. Your code would fail in case like this (which is quite likely):

```
export {
   SomeDependency
   SomeOtherDependency
   SomeThirdDependency,
   SomeFourthDependency,
} from '..................'
```